master/buildbot/process/buildstep.py

```diff
@@ -744,10 +744,16 @@ def start(self):
 
     def interrupt(self, reason):
         self.stopped = True
-        if self._acquiringLock:
-            lock, access, d = self._acquiringLock
-            lock.stopWaitingUntilAvailable(self, access, d)
-            d.callback(None)
+        if self._acquiringLocks:
+            for (lock, access, d) in self._acquiringLocks:
+                if lock.isAvailable(self, access):
+                    # If lock is available to us, we HAVE to claim and release
+                    lock.claim(self, access)
```

Comment:
why? this sounds like a workaround of a more subtle problem

Reply:
It's because we need to remove ourselves from the waiting list. Perhaps the comment could be clarified.

Reply:
Then we should rather add an API to remove ourselves from the waitingList in case of cancel.